arangod/Utils/CollectionNameResolver.cpp

```diff
@@ -278,12 +278,12 @@ std::string CollectionNameResolver::localNameLookup(TRI_voc_cid_t cid) const {
   std::string name;
 
   if (ServerState::isDBServer(_serverRole)) {
-    READ_LOCKER(readLocker, _vocbase->_collectionsLock);
+    READ_LOCKER(readLocker, _vocbase->_dataSourceLock);
+    auto it = _vocbase->_dataSourceById.find(cid);
 
-    auto it = _vocbase->_collectionsById.find(cid);
-
-    if (it != _vocbase->_collectionsById.end()) {
-      if ((*it).second->planId() == (*it).second->cid()) {
+    if (it != _vocbase->_dataSourceById.end()
+        && std::dynamic_pointer_cast<LogicalCollection>(it->second)) {
```

Comment:
please use something like:
```
it->second->category() == LogicalCollection::category()
```
instead of 
```
std::dynamic_pointer_cast<LogicalCollection>(it->second)
```

Reply:
CollectionNameResolver related changes will be done in the next part of task #344